### ISSUE TITLE
write_deltalake fails with error: thread 'tokio-runtime-worker' panicked at 'not implemented'

### ISSUE THREAD

**Issue by jkausti** [delta-io/delta-rs#620]
# Environment

**Delta-rs version**: 0.5.7

**Binding**: python

**Environment**:
- **OS**: Ubuntu 20.04 running in WSL2 in Windows 10
- **Other**: Python 3.8

***
# Bug

**What happened**:
Error message when writing a new deltatable to local filesystem.

I have the following code:
```
import pandas as pd
import deltalake as dl

data = [("a", 1), ("b", 2)]
df = pd.DataFrame(data)
dl.write_deltalake(data=df, table_or_uri="tables/my_table")
```
When run, it generates the following error message:

```
thread 'tokio-runtime-worker' panicked at 'not implemented', rust/src/storage/file/rename.rs:104:25
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
thread '<unnamed>' panicked at 'called `Result::unwrap()` on an `Err` value: JoinError::Panic(...)', rust/src/storage/file/rename.rs:72:14
Traceback (most recent call last):
  File "main.py", line 204, in <module>
    main()
  File "main.py", line 20, in main
    dl.write_deltalake(data=df, table_or_uri="tables/my_table")
  File "/home/john/projects/deltalake-test/.venv/lib/python3.8/site-packages/deltalake/writer.py", line 158, in write_deltalake
    _write_new_deltalake(  # type: ignore[call-arg]
pyo3_runtime.PanicException: called `Result::unwrap()` on an `Err` value: JoinError::Panic(...)
```

However, the delta table gets created even though the error message appears.

**What you expected to happen**:
No error message appears.

**How to reproduce it**:
Run the above code.

**More details**:
I checked the code in [rename.rs](https://github.com/delta-io/delta-rs/blob/8a0475ce14841296b3d68bb962a8cbe5e0fcf2ec/rust/src/storage/file/rename.rs) that the error message refers to and it seems as though the 'not implemented' error happens because of old glibc version (as per the comment on line 103 in rename.rs). However, when running "ldd --version" on my machine, it says my glibc version is 2.31.

**Comment by wjones127** [comment]
Haven't confirmed, but I suspect this is because our Linux wheels have been built in an environment with old glibc for compatibility reasons. I think we'll need to upgrade to manylinux2014, or change that function (I think there might actually be a better approach).

As a workaround, this *should* work if you build from source (annoying, I know):

```bash
git clone https://github.com/delta-io/delta-rs.git
cd delta-rs/python
make install
```

**Comment by jkausti** [comment]
Thanks for the quick response. You are right, I forgot to mention in the bug description that when building it from source the error message does not appear when calling write_deltalake. So that is a workaround for now. Thanks!

**Comment by jkausti** [comment]
This issue was fixed in python release 0.5.8. Closing.
